File: Csharp/303.区域和检索-数组不可变.cs
/*
 * @lc app=leetcode.cn id=303 lang=csharp
 *
 * [303] 区域和检索 - 数组不可变
 */

// @lc code=start
public class NumArray
{
    public List<int> nums_sum = new List<int>() { 0 };

    public NumArray(int[] nums)
    {
        for (var i = 0; i < nums.Length; i++)
        {
            nums_sum.Add(nums_sum[i] + nums[i]);
        }
    }

    public int SumRange(int i, int j)
    {
        return nums_sum[j + 1] - nums_sum[i];
    }

}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(i,j);
 */
// @lc code=end

File: Python3/303.区域和检索-数组不可变.py
#
# @lc app=leetcode.cn id=303 lang=python3
#
# [303] 区域和检索 - 数组不可变
#

# @lc code=start
class NumArray:

    def __init__(self, nums: [int]):
        self.nums = nums

    def sumRange(self, i: int, j: int) -> int:
        numsLen = len(self.nums)
        if j > numsLen-1:
            j = numsLen - 1
        result = 0
        for i in range(i, j+1):
            result += self.nums[i]
        return result

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
# @lc code=end
